### PR TITLE
feat: add request type to obtain IP address used for connecting to rendezvous

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -115,12 +115,46 @@ func (c Client) Discover(ctx context.Context, srv ma.Multiaddr, topic string, li
...
+	return
+}
+
 func (c Client) newStream(ctx context.Context, srv ma.Multiaddr) (rw network.Stream, err error) {
 	pid, err := srv.ValueForProtocol(ethv4.P_ETHv4)
 	if err != nil {
 		return
 	}
-	peerid, err := peer.IDB58Decode(pid)
+	peerid, err := peer.Decode(pid)
```
Comment (jakubgs) [inline client.go:157]: Wait what?

File: server/server_test.go [hunk 2]
```diff
@@ -210,7 +210,7 @@ func TestDiscoverRPC(t *testing.T) {
 				require.Equal(t, protocol.OK, resp.Status)
 			}
 
-			resptype, resp, err := srv.msgParser(protocol.DISCOVER, tc)
+			resptype, resp, err := srv.msgParser(nil, protocol.DISCOVER, tc)
```
Comment (jakubgs) [inline server/server_test.go:213]: Wouldn't we want to include a test for the `protocol.REMOTEIP` command as well?

**Review by jakubgs — APPROVED** [review]
Looks okay to me, aside from the lack of test.

@cammellos  and @Samyoul are out today so we'll have to wait for them to review.

File: client.go (hunk 1 above)
Comment (richard-ramos) [reply client.go:157]: This function was deprecated, and in the latest version of go-libp2p-core it was removed from the source:
https://github.com/libp2p/go-libp2p-core/commit/6cef973b925c54e5028022d3da22e0f73b001541#diff-f53f0da6806dd179092ab3a3013ae56e5eac999b2d78bdc273c2a3ab584b17d4L134-L139

**Comment by jakubgs** [comment]
@cammellos @Samyoul can you check this out?

**Comment by richard-ramos** [comment]
I'm gonna merge this PR as https://github.com/status-im/status-go/pull/2333 was approved

**Pull request merged**